src/shogun/lib/View.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Wuwei Lin
+ */
+#ifndef _VIEW__H__
+#define _VIEW__H__
+
+#include <shogun/features/Features.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/lib/SGVector.h>
+#include <type_traits>
+
+namespace shogun
+{
+
+	template <class T>
+	T* view(T* viewable, const SGVector<index_t>& subset)
+	{
+		static_assert(
+		    std::is_base_of<CFeatures, T>::value ||
+		        std::is_base_of<CLabels, T>::value,
+		    "Only CFeatures and CLabels are viewable.");
```

Comment:
I am not a fan of error messages that contain things that might change in the future, i.e. a new class is made viewable.
I would just state: Class is not viewable. The compiler error will provide the T and also the static assert which will tell the caller what is viewable and what is not. Or?